src/main/java/spinner/sudoku/SudokuGui.java

```diff
@@ -0,0 +1,137 @@
+package spinner.sudoku;
+
+import javax.swing.*;
+import java.awt.*;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.util.ArrayList;
+import java.util.List;
+
+/** Plan for sudokuGUI
+ * GUI will be a 9x9 grid of JTextField components
+ * Will display a partially completed board
+ * User can manually enter numbers into the cells
+ *
+ * GUI will interact with the Sudoku logic to validate the board and retrieve errors.
+ */
+
+public class SudokuGui {
+    private JFrame frame;
+    private JTextField[][] cells;
+    private Sudoku sudoku;
+
+    public SudokuGui(Sudoku sudoku) {
+        this.sudoku = sudoku;
+        initializeGui();
+    }
+
+    private void initializeGui() {
+        frame = new JFrame("Sudoku");
+        frame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
+        frame.setSize(600, 650);
+
+        JPanel gridPanel = new JPanel();
+        gridPanel.setLayout(new GridLayout(9, 9));
+
+        cells = new JTextField[9][9];
+
+        int[][] board = sudoku.getBoard();
+
+        for (int i = 0; i < 9; i++) {
+            for (int j = 0; j < 9; j++) {
+                cells[i][j] = new JTextField();
+                cells[i][j].setHorizontalAlignment(JTextField.CENTER);
+
+                if (board[i][j] != 0) { // Pre-filled numbers
+                    cells[i][j].setText(String.valueOf(board[i][j]));
+                    cells[i][j].setEditable(false);
+                    cells[i][j].setBackground(Color.LIGHT_GRAY);
+                }
+
+                gridPanel.add(cells[i][j]);
+            }
+        }
+
+        JButton checkButton = new JButton("Check Errors");
+
+        // Updated action listener to process errors programmatically
+        checkButton.addActionListener(new ActionListener() {
```

Comment:
I wanted you to do it without a button. 
You should check the board whenever a user puts in a number